File: c/readability/readability.c
#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

int main(void)
{
    // getting text from user
    string text = get_string("Text: ");

    // counting letter, spaces and words
    int letters = 0;
    int words = 1;
    int sentences = 0;
    for (int i = 0, l = strlen(text); i < l; i++)
    {
        if (text[i] != ' ' &&
            ((text[i] >= 'A' && text[i] <= 'Z') || (text[i] >= 'a' && text[i] <= 'z')))
        {
            letters += 1;
        }
        if (text[i] == ' ')
        {
            words += 1;
        }
        if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            sentences += 1;
        }
    }
    // printf("letters: %i, words: %i, sentences: %i\n", letters, words, sentences);

    // computer grade
    int grade = (int) round(
        (0.0588 * (((float) letters / words) * 100) - 0.296 * (((float) sentences / words) * 100)) -
        15.8);

    // print grade
    if (grade < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (grade > 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", grade);
    }
}

File: python/sentimental-readability/readability.py
text = str(input("Text: "))

letters = 0
words = 0
sentences = 0

for i in range(len(text)):
    if text[i].isalpha():
        letters += 1
    if text[i] == " ":
        words += 1
    if text[i] == "." or text[i] == "!" or text[i] == "?":
        sentences += 1

words += 1

L = (letters / words) * 100
S = (sentences / words) * 100

gradeLevel = (0.0588 * L) - (0.296 * S) - 15.8

if gradeLevel < 1:
    print("Before Grade 1")
elif gradeLevel >= 16:
    print("Grade 16+")
else:
    print(f"Grade {round(gradeLevel)}")
